src/components/EnterpriseApp/data/enterpriseCurationReducer.js

```diff
@@ -43,18 +48,59 @@ export const enterpriseCurationActions = {
     type: ADD_HIGHLIGHT_SET,
     payload,
   }),
+  setEnterpriseHighlightedContents: (payload) => ({
+    type: SET_ENTERPRISE_HIGHLIGHTED_CONTENTS,
+    payload,
+  }),
+  setIsNewArchivedCourse: (payload) => ({
+    type: SET_IS_NEW_ARCHIVED_COURSE,
+    payload,
+  }),
+  updateDismissedArchivedCourse: (payload) => ({
+    type: UPDATE_DISMISSED_ARCHIVED_COURSE,
+    payload,
+  }),
 };
 
 function getHighlightSetsFromState(state) {
   return state.enterpriseCuration?.highlightSets || [];
 }
 
+function getIsNewArchivedCourseFromState(payload) {
+  const dismissedCookies = archivedHighlightsCoursesCookies.get(NEW_ARCHIVED_COURSE_ALERT_DISMISSED_COOKIE_NAME);
+  const isDismissedArchivedCoursesSet = dismissedCookies !== undefined;
+  const dismissedArchivedCourses = isDismissedArchivedCoursesSet ? dismissedCookies : {};
+  let isNewArchivedCourse = false;
+  // checks that the highlight uuid isn't set
+  payload?.forEach(highlightSet => {
+    if (!dismissedArchivedCourses[highlightSet.uuid]) {
+      highlightSet.highlightedContent.forEach(courseContent => {
+        // if the length is greater than 1 for the course run status e.g. ["archived", "published"]
+        // the course should not be archived
+        if (courseContent.courseRunStatuses?.length === 1 && courseContent.courseRunStatuses?.includes('archived')) {
```

Comment:
[curious] What about a situation where there's an archived course run and an unpublished course run:

```js
courseContent.courseRunStatuses = ['archived', 'unpublished']
```

Given that the the unpublished course run is effectively non-existent to learners, would the only relevant course run status remaining be archived, in which `isNewArchivedCourse` might be truthy in this case? 🤔

Reply:
In this scenario, we wouldn't want to consider the content as archived because an "unpublished" status could have an upcoming scheduled course run.